base/compiler/ssair/ir.jl

```diff
@@ -546,30 +546,58 @@ mutable struct IncrementalCompact
             domtree = construct_domtree(blocks)
             for i = 1:length(bb_rename)
                 if bb_unreachable(domtree, i)
-                    bb_rename[i] = -1
+                    bb_rename[i] = -1 # Dead block
                 else
-                    bb_rename[i] = cur_bb
-                    cur_bb += 1
+                    preds = blocks[i].preds
+                    # Is this block the sole successor of it's sole predecessor?
+                    # If yes we can merge them.
+                    if length(preds) == 1 && length(blocks[first(preds)].succs) == 1
```

Comment:
This doesn't quite work. You can only do this if the three blocks in question are sitting in statement order or if the block in question only has one statement. You need to check those conditions here. Anything that needs to rearrange instructions or rewrite phi nodes is too complicated for IncrementalCompact.